src/vaadin-context-menu.html

```diff
@@ -566,7 +566,14 @@
           // Then set styling attrs for flex-aligning the content appropriatelly.
           const wdthVport = document.documentElement.clientWidth;
           const hghtVport = document.documentElement.clientHeight;
-          if (x < wdthVport / 2 || x < xMax) {
+
+          const parent = overlay.parentOverlay;
+          if (parent && parent.hasAttribute('right-aligned')) {
```

Comment:
Wouldn't this rule always right-align every descendant sub-menu if one of the menus happens to be right-aligned?

Reply:
Yes, after taking a deeper look I think there should be a better fix. Will continue working on it.

Reply:
Updated and added a test to reproduce this case, see it also in the demo:
<img width="501" alt="Screen Shot 2019-03-25 at 14 27 00" src="https://user-images.githubusercontent.com/10589913/54919694-29bc8a00-4f0a-11e9-8ac8-949ead90ab96.png">